### PR TITLE
Add Welcome component

### PR DESCRIPTION
**Pull request by r-czajkowski**
Closes: #824 
Closes: AENG-1

This PR adds a `Welcome` component based on the [Figma views](https://www.figma.com/design/Ov4M1oVlSrO4Pl7NmGOHC1/Ledger-2025?node-id=446-3007&node-type=frame&t=Oo0Jpv4rJCIZkXqq-0). Here we are also updating how we display modal in the correct order - we use react router loaders to display modals in the correct order. We want to display the welcome component after the access code modal in a standalone app and before the wallet connection in embedded apps.

<details>
  <summary>Screenshots</summary>
  
  obraz

obraz

obraz
</details>

### CODE REVIEW

File: dapp/src/assets/videos/step-1.mov
Comment (r-czajkowski) [inline dapp/src/assets/videos/step-1.mov:1]: MOV format is not supported in Firefox we need to ask Soring to export videos in mp4.

File: dapp/src/components/WelcomeModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        </Box>
+        <Box
+          as="video"
+          src={activeStepData.video}
+          width="320px"
+          height="100%"
+          autoPlay
+          muted
+          objectFit="cover"
+          borderRadius="xl"
```
Comment (kpyszkowski) [inline dapp/src/components/WelcomeModal/index.tsx:167]:
```suggestion
          rounded="xl"
```

File: dapp/src/components/WelcomeModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  ModalFooter,
+  useSteps,
+  SimpleGrid,
+  StepIndicatorProps,
+  UseStepsReturn,
+} from "@chakra-ui/react"
+import { H3, TextSm } from "#/components/shared/Typography"
+import { BaseModalProps } from "#/types"
+import withBaseModal from "../ModalRoot/withBaseModal"
+import step1Video from "../../assets/videos/step-1.mov"
```
Comment (kpyszkowski) [inline dapp/src/components/WelcomeModal/index.tsx:20]: For ease of exports and clarity we can organize these video assets by purpose so `/assets/videos/welcome-steps/welcome-step-{n}.mov` with `index.ts` file. Please take a look at `/assets/images/benefits` as a reference.

**Review by kpyszkowski — DISMISSED** [review]
Left review. Overall looks good. Please use style props shorthands (eg. `w` instead of `width`) and style tokens (eg. `2.5` instead of `"10px"`) if possible (indicated below).

File: dapp/src/components/WelcomeModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+        />
+      </SimpleGrid>
+    </>
+  )
+}
+
+const WelcomeModal = withBaseModal(WelcomeModalBase, {
+  // TODO: Define the new size for welcome modal or update `xxl` size values.
+  size: "xl",
+  border: "none",
```
Comment (kkosiorowska) [inline dapp/src/components/WelcomeModal/index.tsx:177]: Designs have changed and now none of the modals have a border.  So I think we can disable the border for all modal windows by default in the theme.

File: dapp/src/components/ModalRoot/withBaseModal.tsx [hunk 4]
```diff
@@ -1,13 +1,21 @@
...
 import { BaseModalProps } from "#/types"
 import { useAppNavigate, useSidebar } from "#/hooks"
 
 const MODAL_BASE_SIZE = "lg"
 
 function withBaseModal<T extends BaseModalProps>(
   WrappedModalContent: ComponentType<T>,
-  modalProps?: Omit<ModalProps, "isOpen" | "onClose" | "children">,
+  modalProps?: Omit<ModalProps, "isOpen" | "onClose" | "children"> & {
+    border?: BoxProps["border"]
```
Comment (kkosiorowska) [inline dapp/src/components/ModalRoot/withBaseModal.tsx:17]: As I mentioned in another comment, we can simply disable the border for all modal windows.

File: dapp/src/components/WelcomeModal/index.tsx
Comment (kkosiorowska) [inline dapp/src/components/WelcomeModal/index.tsx:1]: We should reduce the directory structure and pull up existing components one level higher, if possible. Please see this [thread](https://discord.com/channels/1079490158639976609/1209922628966416474/1210146641097785395).

File: dapp/src/assets/videos/step-1.mov
Comment (r-czajkowski) [reply dapp/src/assets/videos/step-1.mov:1]: @SorinQ could you please export videos as `mp4` files?

File: dapp/src/components/WelcomeModal/index.tsx (hunk 2 above)
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 7da4627

File: dapp/src/components/WelcomeModal/index.tsx (hunk 1 above)
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+            >
+              {isLastStep ? "Get started" : "Skip"}
+            </Button>
+          </ModalFooter>
+        </Box>
+        <Box
+          as="video"
+          src={activeStepData.video}
+          width="320px"
+          height="100%"
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+              onClick={isLastStep ? closeModal : goToNext}
+            >
+              {isLastStep ? "Get started" : "Skip"}
+            </Button>
+          </ModalFooter>
+        </Box>
+        <Box
+          as="video"
+          src={activeStepData.video}
+          width="320px"
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+        One dashboard{" "}
+        <Box as="span" color="orange.30">
+          endless rewards
+        </Box>
+      </H3>
+    ),
+    content: (
+      <>
+        As a depositor, you&apos;re automatically in the{" "}
+        <Box as="strong">Acre Points Program</Box>. Enjoy daily points drops and
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+      w: "15px",
+      borderRadius: "6",
+      _after: {
+        content: '""',
+        position: "absolute",
+        opacity: "0.15",
+        w: "15px",
+        h: "10px",
+        borderRightRadius: "5",
+        left: "6px",
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+    ),
+    video: step3Video,
+  },
+]
+
+const stepIndicatorStyleProps: StepIndicatorProps = {
+  sx: {
+    "[data-status=active] &": {
+      opacity: 1,
+      w: "15px",
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+      borderRadius: "6",
+      _after: {
+        content: '""',
+        position: "absolute",
+        opacity: "0.15",
+        w: "15px",
+        h: "10px",
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+  sx: {
+    "[data-status=active] &": {
+      opacity: 1,
+      w: "15px",
+      borderRadius: "6",
+      _after: {
+        content: '""',
+        position: "absolute",
+        opacity: "0.15",
+        w: "15px",
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+        left: "6px",
+        bgColor: "orange.50",
+      },
+    },
+    "&[data-status=complete], [data-status=incomplete] &": {
+      bgColor: "orange.50",
+    },
+  },
+  border: "none",
+  w: "10px",
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: 1197034

File: dapp/src/components/WelcomeModal/index.tsx [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+      <H3 fontWeight="semibold">
+        <Box as="span" color="orange.30">
+          Battle-tested{" "}
+        </Box>
+        in the market
+      </H3>
+    ),
+    content: (
+      <>
+        Acre is powered by <Box as="strong">tBTC</Box>, the trusted Bitcoin
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal/index.tsx]: b5fc3b7

File: dapp/src/theme/utils/colors.ts [hunk 14]
```diff
@@ -73,5 +73,20 @@ export const colors = {
       "05": "rgba(0, 0, 0, 0.05)",
       15: "rgba(0, 0, 0, 0.15)",
     },
+    orange: {
+      30: {
+        50: {},
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/theme/utils/colors.ts]: 1cb59fa

File: dapp/src/assets/videos/step-1.mov
Comment (r-czajkowski) [reply dapp/src/assets/videos/step-1.mov:1]: 10562fc

**Review by kpyszkowski — COMMENTED** [review]
LGTM 🔥

File: dapp/src/pages/AccessPage/loader.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+export const redirectToAccessPageLoader = async (
+  url: string,
+  redirectToOnSuccess: null | string = null,
+  redirectToOnFail: null | string = null,
+) => {
+  try {
+    const isValid = await acreApi.verifyAccessCode(
+      getAccessCodeFromLocalStorage() ?? "",
+    )
```
Comment (kkosiorowska) [inline dapp/src/pages/AccessPage/loader.ts:14]: Even if `encodedCode` is not defined we send a request to `acreApi`.

```suggestion
    const encodedCode = getAccessCodeFromLocalStorage()
    const isValid = encodedCode
      ? await acreApi.verifyAccessCode(encodedCode)
      : false
```

**Review by kkosiorowska — COMMENTED** [review]
Works great 🚀 I left some minor comments.

File: dapp/src/components/WelcomeModal.tsx [hunk 16]
```diff
@@ -1,83 +1,194 @@
...
+  position: "relative",
+  opacity: "0.15",
+}
+
+function WelcomeModalBase({ closeModal }: BaseModalProps) {
+  // Cast to fix eslint error: `unbound-method`.
+  const { activeStep, goToNext } = useSteps({
+    index: 0,
+    count: steps.length,
+  }) as UseStepsReturn & { goToNext: () => void }
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/components/WelcomeModal.tsx:135]: Yeah. But it's not a good time to update main packages just before the release.

File: dapp/src/hooks/useInitApp.ts [hunk 17]
```diff
@@ -7,15 +7,13 @@ import useDetectEmbed from "./useDetectEmbed"
...
 import { useFetchBTCPriceUSD } from "./useFetchBTCPriceUSD"
-import useGatingDApp from "./useGatingDApp"
 
 export function useInitApp() {
   // TODO: Let's uncomment when dark mode is ready
   // useDetectThemeMode()
   useSentry()
   useDetectEmbed()
   useDetectReferral()
-  useGatingDApp()
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/hooks/useInitApp.ts:18]: 3b192c4

File: dapp/src/theme/Modal.ts [hunk 18]
```diff
@@ -13,6 +13,7 @@ const baseStyleDialog = defineStyle({
   borderColor: "white",
   borderRadius: "xl",
   bg: "gold.100",
+  border: "none",
```
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/theme/Modal.ts:14]: de3545a

File: dapp/src/router/utils.ts
[the remark this answers is not in the document]
Comment (r-czajkowski) [reply dapp/src/router/utils.ts:1]: fa4943

File: dapp/src/pages/AccessPage/loader.ts (hunk 15 above)
Comment (r-czajkowski) [reply dapp/src/pages/AccessPage/loader.ts]: b98b7bf

File: dapp/src/components/WelcomeModal.tsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (nkuba) [reply dapp/src/components/WelcomeModal.tsx:134]: Let's add it to our backlog.

**Review by kkosiorowska — DISMISSED** [review]
LGTM 🚀

**Review by kkosiorowska — APPROVED** [review]

**Pull request merged**